dialog_lib/db/session.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session, sessionmaker
+
+from contextlib import contextmanager
+
+engine = sa.create_engine(os.environ.get("DATABASE_URL"))
+Session = sessionmaker(bind=engine)
+
+@contextmanager
+def get_session():
+    session = Session()
+    try:
+        yield session
+        session.flush()
+        session.commit()
+    except Exception as e:
+        session.rollback()
+        raise e
+    finally:
+        session.close()
```

Comment:
```suggestion
@contextmanager
def get_session():
    with Session() as session:
        try:
            yield session
            session.commit()
        except Exception as exc:
            session.rollback()
            raise exc
```

1. The `Session` instance can be used as a context manager; it calls the `.close()` method for you on `__exit__`.
2. The `.commit()` operation unconditionally issues `.flush()` before emitting COMMIT